File: Queue and Stack/MinStack.py
class MinStack:
	def __init__(self):
		self.stack = []
		self.min = []
		self.length = 0

	def push(self, val):
		self.stack = self.stack + [val]
		if self.length < 1:
			self.min = self.min + [val]
		elif val < self.min[self.length-1]:
			self.min = self.min + [val]
		else:
			self.min = self.min + [self.min[self.length-1]]
		self.length += 1

	def pop(self):
		self.min.pop()
		pop_ele = self.stack.pop()
		self.length -= 1
		return pop_ele

	def peek(self):
		peek_ele = self.stack[self.length-1]
		return peek_ele

	def minVal(self):
		min_ele = self.min[self.length-1]
		return min_ele

File: Queue and Stack/MinStack.js
function MinStack(){
	this.stack = [];
	this.min = [];
	this.len = 0;
}

MinStack.prototype.add = function(val){
	this.stack.push(val);
	if(this.len < 1){
		this.min.push(val);
	} else{
		this.min.push(Math.min(val, this.min[this.len-1]));
	}
	this.len++;
}

MinStack.prototype.delete = function(){
	this.stack.pop();
	this.min.pop();
	this.len--;
}

MinStack.prototype.minVal = function(){
	return this.min[this.len-1];
}

var ms = new MinStack();
ms.add(2);
ms.add(3);
ms.add(4);
ms.add(5);

console.log(ms.minVal());
ms.delete();
ms.delete();
ms.add(1);
console.log(ms.minVal());
ms.add(0);
console.log(ms.minVal());
ms.delete();
console.log(ms.minVal());
ms.delete();
console.log(ms.minVal());
ms.delete();
ms.delete();